spec/fenrir_view/property_types_spec.rb

```diff
@@ -0,0 +1,299 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe FenrirView::PropertyTypes do
+  let(:mock_instance) {
+    lambda { |instance_props, component_props|
+      FenrirView::PropertyTypes.new({
+                                      component_class: String,
+                                      component_properties: component_props,
+                                      instance_properties: instance_props,
+                                    })
+    }
+  }
+
+  let(:mock_component) {
+    lambda { |content, validations|
+      FenrirView::PropertyTypes.new({
+                                      component_class: String,
+                                      component_properties: {},
+                                      instance_properties: {},
+                                    }).validate_property(:title, content, validations)
+    }
+  }
+
+  let(:title_property) { 'The best title' }
+
+  describe 'Presenter initializes validator' do
+    let(:card_facade_property_types) {
+      FenrirView::Presenter.component_for('component', 'card', {
+        title: 'everything',
+      }, validate: true).send(:property_types)
+    }
+
+    let(:component_properties) do
+      {
+        yield: {},
+        title: { required: true },
+        description: {},
+        link: {},
+        image_url: {},
+        location: {},
+        data: { default: [] }
+      }
+    end
+
+    let(:instance_properties) do
+      {
+        yield: nil,
+        title: 'everything',
+        description: nil,
+        link: nil,
+        image_url: nil,
+        location: nil,
+        data: []
+      }
+    end
+
+    it 'Presenter can initialize validator' do
+      expect(card_facade_property_types).to respond_to(:validate_properties)
+
+      expect(card_facade_property_types.instance_variable_get(:@component_class)).to eq(CardFacade)
+      expect(card_facade_property_types.instance_variable_get(:@component_properties)).to eq(component_properties)
+      expect(card_facade_property_types.instance_variable_get(:@instance_properties)).to eq(instance_properties)
+    end
+  end
+
+  describe '#validate_properties' do
+    it 'returns instance properties when correct' do
+      expect(mock_instance[{ title: title_property }, { title: {}}].validate_properties).to eq({ title: title_property })
+    end
+
+    it 'returns nil if instance is protected by guard clause' do
+      expect(mock_instance[false, { title: {}}].validate_properties).to be_nil
+    end
+
+    it 'raises error if properties is not a hash' do
+      expect { mock_instance[[1,2,3], { title: {}}].validate_properties }.to raise_error('An instance of String has properties as Array. Should be Hash or false.')
```

Comment:
These error messages in the tests aren't as useful as they could be because I'm mocking the Component Facade Class as `String`. I should change it to an actual component class.